### PR TITLE
[FIX] Allow default values for payment mode in multi-company env

### PR DESCRIPTION
**Pull request by CasVissers-360ERP**
If using ir.default for payment mode in a multi-company environment the payment mode of a sister company could be loaded if superuser (uid 1) was active in this sister company. Resulting in the error message:
_("The company of the invoice %s does not match with that of the payment mode") % rec.name)

This PR fixes this issue.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please indicate steps to reproduce the problem. I don't think populating `company_id` in the refund fields is the solution to your problem.

**Comment by CasVissers-360ERP** [comment]
@pedrobaeza I would like to create a runbot video, but OCA runbot seems not accessible?

**Comment by pedrobaeza** [comment]
You can use the "Try me on runbot" button. Any way, I still prefer written steps than a hard to follow video. And as said, a technical explanation why this solves the problem is also welcome (included as comment/docstring of the method as well for future references).
